src/txn/txn_timestamp.c

```diff
@@ -273,6 +273,73 @@ __txn_global_query_timestamp(
 		 */
 		if (ts == 0)
 			return (WT_NOTFOUND);
+	} else if (WT_STRING_MATCH("all_durable", cval.str, cval.len)) {
```

Comment:
Any opinions on how I've done this?

I first check the queue of transactions ordered by durable timestamp which contains only transactions that explicitly have a durable timestamp set. Then I go and check the queue ordered by commit timestamp and look for the first one that doesn't have a durable timestamp (and is therefore non-prepared).

The other alternative is to put everything in the durable queue and have it ordered by `x` where `x` is `durable_timestamp` for prepared transactions and `commit_timestamp` for non-prepared transactions. That way, I can just check the one queue and avoid multiple rounds of locking. I didn't go for that approach since it seems more troublesome to maintain the queues and that setting a commit_timestamp will mean locking and inserting into two different queues.

Reply:
Do we need the commit timestamp queue for anything other than answering "all_committed" queries?  I'm assuming that will go away once we've updated MongoDB to use "all_durable"...

Reply:
Yeah thanks for pointing that out. The commit timestamp queue is there purely for `all_committed`. In that case, it makes sense to not rely on that queue so it can be removed later.

Reply:
713350c

Again, I'm unsure about a lot of this code.
So if you could take a detailed look that'd be great.